components/message.php

```diff
@@ -0,0 +1,504 @@
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message create --from=user1 --to=user2 --subject="Message Title" --content="We are ready"
+	 *     Success: Message successfully created.
+	 *
+	 *     $ wp bp message create --from=545 --to=313
+	 *     Success: Message successfully created.
+	 *
+	 * @alias add
+	 */
+	public function create( $args, $assoc_args ) {
+		$r = wp_parse_args( $assoc_args, array(
+			'to'        => '',
+			'subject'   => '',
+			'content'   => '',
+			'thread-id' => false,
+			'date-sent' => bp_core_current_time(),
+			'silent'    => false,
+		) );
+
+		// Fill in the content.
+		if ( empty( $r['content'] ) ) {
+			$r['content'] = $this->generate_random_text();
+		}
+
+		// Fill in the subject.
+		if ( empty( $r['subject'] ) ) {
+			$r['subject'] = sprintf( 'Message Subject' );
+		}
+
+		$user = $this->get_user_id_from_identifier( $assoc_args['from'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		// To is not required when thread id is set.
+		if ( ! empty( $r['to'] ) ) {
+			$recipient = $this->get_user_id_from_identifier( (int) $r['to'] );
+
+			if ( ! $recipient ) {
+				WP_CLI::error( 'No user found by that username or ID.' );
+			}
+		}
+
+		// Existing thread recipients will be assumed.
+		$recipient = ( ! empty( $r['thread_id'] ) ) ? array() : array( $recipient->ID );
+
+		$thread_id = messages_new_message( array(
+			'sender_id'  => $user->ID,
+			'thread_id'  => $r['thread-id'],
+			'recipients' => $recipient,
+			'subject'    => $r['subject'],
+			'content'    => $r['content'],
+			'date_sent'  => $r['date-sent'],
+		) );
+
+		if ( ! is_numeric( $thread_id ) ) {
+			WP_CLI::error( 'Could not add a message.' );
+		}
+
+		if ( $r['silent'] ) {
+			return;
+		}
+
+		if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'porcelain' ) ) {
+			WP_CLI::line( $thread_id );
+		} else {
+			WP_CLI::success( 'Message successfully created.' );
+		}
+	}
+
+	/**
+	 * Delete message thread(s) for a given user.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <thread-id>...
+	 * : Thread ID(s).
+	 *
+	 * --user-id=<user>
+	 * : Identifier for the user. Accepts either a user_login or a numeric ID.
+	 *
+	 * [--yes]
+	 * : Answer yes to the confirmation message.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp bp message delete 500 687867 --user-id=40
+	 *     Success: Thread successfully deleted.
+	 *
+	 *     $ wp bp message delete 564 5465465 456456 --user-id=user_logon --yes
+	 *     Success: Thread successfully deleted.
+	 *
+	 * @alias remove
+	 */
+	public function delete( $args, $assoc_args ) {
+		$thread_id = $args[0];
+
+		$user = $this->get_user_id_from_identifier( $assoc_args['user-id'] );
+		if ( ! $user ) {
+			WP_CLI::error( 'No user found by that username or ID.' );
+		}
+
+		WP_CLI::confirm( 'Are you sure you want to delete this thread(s) ?', $assoc_args );
+
+		parent::_delete( array( $thread_id ), $assoc_args, function( $thread_id ) {
+
+			if ( messages_delete_thread( $thread_id, $user->ID ) ) {
```

Comment:
`messages_delete_thread()` is pretty weird - it returns true if you pass `123` as the thread ID, even if `123` is not a valid thread ID. I got confused about this in combination with the `delete-thread` comment above - the CLI command said "success" even though nothing had been deleted.

I think we can probably fix this upstream, by getting `messages_delete_thread()` to return false in various cases (where the thread ID is not a thread; where `$user_id` doesn't match a thread member; where the deletion fails for some other reason) and then inherit it here. A workaround in wp-cli-buddypress seems like the wrong fix. Do you agree? If so, perhaps you could open a BP ticket?